Defer checking for arc4random and friends until libcrypto is found,
fixes segv due to portable libressl/opensmtpd and arc4random<->RAND_bytes.

See http://marc.info/?l=openssh-unix-dev&m=140515171616098&w=2
for more information.
--- configure.ac.orig	2014-08-03 18:23:43.352548006 +0200
+++ configure.ac	2014-08-03 18:24:00.440690901 +0200
@@ -620,9 +620,6 @@ fi
 #l1527 (customized)
 dnl    Checks for library functions. Please keep in alphabetical order
 AC_CHECK_FUNCS([ \
-	arc4random \
-        arc4random_buf \
-        arc4random_uniform \
 	asprintf \
 	b64_ntop \
 	__b64_ntop \
@@ -1369,6 +1366,8 @@ AC_RUN_IFELSE(
 	]
 )
 
+AC_CHECK_FUNCS([arc4random arc4random_buf arc4random_uniform])
+
 # PRNGD TCP socket
 AC_ARG_WITH([prngd-port],
 	[  --with-prngd-port=PORT  read entropy from PRNGD/EGD TCP localhost:PORT],
